README.md

```diff
@@ -68,58 +64,332 @@ To build all examples via gradle, simply call:
+
+In the following a short overview is given on each component.
+
+
+
+  1. **The MPS Language**
+
+     The language (meta-model) and model used in all examples.
+
+     [See the 'Language' section in the MPS README.md](mps/README.md#language)
+
+
+  2. **Generated model API (domain API)**
+
+     An API based on the meta-model generated with the [api-gen](https://github.com/modelix/api-gen) plugin from modelix.
+
+     [See the 'Generated API' section in the MPS README.md](mps/README.md#generated-api)
+
+  3. **Domain-specific openAPI**
+
+     In this example project, an extra domain-specific API layer is added which is defined in the [openAPI](https://www.openapis.org/) [specification](/openapi).
+     This layer is meant educational as there are no noteworthy abstractions happen in this definition.
+     It intends to show how one introduces a clearly defined domain-specific abstraction isolating the language engineering (meta-modeling) and the web development.
+
+     We provide two implementations of the API layer: The [`rest-api-json-bulk`](/rest-api-json-bulk) and [`rest-api-model-server`](/rest-api-model-server) components.
+
+     For more details, also see the ['OpenAPI of the Courses domain' section in the MPS README.md](openapi/README.md)
+
+
+
+  4. **openAPI implementation**
+
+     This project provides two implementations of the [openAPI](openapi) domain abstraction.
+
+     - A. MPS as a source (`rest-api-json-bulk` component)
+
+          This API implementation provides access to the model by obtaining the model knowledge directly from a running MPS instance.
+          It is implemented using ktor and connects to the [`json-bulk-model-access`](https://github.com/modelix/mps-rest-model-access) plugin running inside of MPS.
+          This component can only provide **read only access** as the `json-bulk-model-access` is read only.
+
+          For more details, also see the [`rest-api-json-bulk` README.md](rest-api-json-bulk/README.md) for details.
+
+
+
+     - B. model-server as a source (`rest-api-model-server` component)
+
+          This API implementation provides access to the model by connecting to a running [`modelix model-server`](https://github.com/modelix/modelix.core/tree/main/model-server).
+          It is implemented using Quarkus and can provide **read an write access** to the underlying model.
+          This is realized using websockets exposed by the `model-server`.
+
+          For more details, also see the [`rest-api-model-server` README.md](/rest-api-model-server/README.md) for details.
+
+
+
+
+  5. **Single-page application (SPA) Dashboard**
 
-### 1. The MPS Language
+     The dashboard provides access to model knowledge through a browser.
+     As it is conforming to the openAPI specification, the dashboard is able to obtain the model content from both openAPI implementations.
```

Comment:
```suggestion
     As it is conforming to the OpenAPI specification, the dashboard is able to obtain the model content from both backen implementations.
```

OpenAPI implementations is misleading here. Backend and frontend implement the spec. Maybe we should somehow call the Quarkus and ktor applications "backends" or something else and use this term consistently throughout the documentation?